docs/development/docs/code-style.md

```diff
@@ -0,0 +1,128 @@
+  typing related classes like `t.TypedDict`.
+
+      Use the new syntax and classes for typing introduced with Python 3.10 and
+      available using `from __future__ import annotations` since Python 3.8. Be
+      aware however that this only works in the context of annotations; the code
+      still needs to run on Python 3.8! This means that in some (rare) cases, you
+      *must* use the old-style type hints.
+
+      - Instead of `t.Tuple`, `t.List` etc. use the builtin classes `tuple`, `list`
+        etc.
+      - For classes that are not builtin (e.g. `Iterable`), `import collections.abc
+        as cabc` and then use them like `cabc.Iterable`.
+      - Use [PEP-604]-style unions, e.g. `int | float` instead of `t.Union[int, float]`.
+      - Use `... | None` (with `None` always as the last union member) instead of
+        `t.Optional[...]` and always explicitly annotate where `None` is possible.
+
+- **Python style rules**: For conflicting parts, the [black] code style
+  wins. If you have set up `black` correctly, you don't need to worry about
+  this though :)
+- When working with `dict`s, consider using `t.TypedDict` instead of a more
+  generic `dict[str, float|int|str]`-like annotation where possible, as the
+  latter is much less precise (often requiring additional `assert`s or
+  `isinstance` checks to pass) and can grow unwieldy very quickly.
+- Prefer `t.NamedTuple` over `collections.namedtuple`, because the former uses
+  a more convenient `class ...:` syntax and also supports type annotations.
+
+## Naming Conventions
+
+### Database Models
+
+All SQLAlchemy model should have `Database` as a prefix, e.g., `DatabaseProject` or
+`DatbaseUser`.
+
+# Frontend
+
+## General
+
+## Naming Conventions
+
+### Crud Functions
+
+#### Creating resource
+
+For creating a resource one should use `create` as a prefix followed by
+the resource one wants to create (e.g., `createProject(...): Observable<Project>`)
+
+#### Retriving resources
+
+In general, one should use `get` as prefix for a function that retrieves and
+returns resources.
+In case, one wants to get a list of alls resources use `get + <resource type> + s`
+as function name (e.g., `getProjects`).
+In case, one wants to get a specific resource use `get + <resource type> + By + <property>`
+as function name (e.g., `getProjectByName(name: string)`). The only exception
+to this rule is the case where one wants to get a resource by its identifier
+where the `By + <property>` part can be omitted (e.g., `getProject(id: number)`).
+
+In some services, we use the concept of having a static representation of a resource or
+a list of resources as observable inside the service. For example,
+
+```ts
+private _gitModel = new Subject<GetGitModel>();
+public _gitModels = new BehaviorSubject<Array<GetGitModel>>([]);
+
+readonly gitModel = this._gitModel.asObservable();
+readonly gitModels = this._gitModels.asObservable();
+```
```

Comment:
```suggestion
```ts
private _gitModel = new Subject<GetGitModel>();
private _gitModels = new BehaviorSubject<Array<GetGitModel>>([]);

readonly gitModel = this._gitModel.asObservable();
readonly gitModels = this._gitModels.asObservable();
```
```